dashboard/src/actions/overviewActions.js

```diff
@@ -0,0 +1,201 @@
+      const initNewRuns = getState().overview.initNewRuns;
+
+      const method = actions[actionType];
+
+      const dataIndex = newRuns.findIndex(
+        (item) => item.resource_id === dataset.resource_id
+      );
+      const endpoints = getState().apiEndpoint.endpoints;
+      const response = await API.put(
+        `${endpoints?.api?.datasets_metadata}/${dataset.resource_id}`,
+        {
+          metadata: { [method]: updateAction },
+        }
+      );
+      if (response.status === 200) {
+        if (actionType === "save") {
+          savedRuns.push(dataset);
+          const filteredNewRuns = newRuns.filter(
+            (item) => item.resource_id !== dataset.resource_id
+          );
+          const filteredInitRuns = initNewRuns.filter(
+            (item) => item.resource_id !== dataset.resource_id
+          );
+          dispatch({
+            type: TYPES.SAVED_RUNS,
+            payload: savedRuns,
+          });
+          dispatch({
+            type: TYPES.INIT_NEW_RUNS,
+            payload: filteredInitRuns,
+          });
+          dispatch({
+            type: TYPES.NEW_RUNS,
+            payload: filteredNewRuns,
+          });
+        } else {
+          newRuns[dataIndex].metadata[actions[actionType]] =
+            response.data[actions[actionType]];
+          dispatch({
+            type: TYPES.NEW_RUNS,
+            payload: newRuns,
+          });
+        }
+      } else {
+        dispatch(constructToast("danger", response?.data?.message));
+      }
+    } catch (error) {
+      dispatch(constructToast("danger", error?.response?.data?.message));
+      dispatch({ type: TYPES.NETWORK_ERROR });
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  };
+
+export const deleteDataset = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    const response = await API.post(
+      `${endpoints?.api?.datasets_delete}/${dataset.resource_id}`
+    );
+    if (response.status === 200) {
+      const datasets = getState().overview.newRuns;
+      const initNewRuns = getState().overview.initNewRuns;
+      const result = datasets.filter(
+        (item) => item.resource_id !== dataset.resource_id
+      );
+      const filteredInitRuns = initNewRuns.filter(
+        (item) => item.resource_id !== dataset.resource_id
+      );
+      // dispatch is awaited else toast message will appear before the UI updated
+      await dispatch({
+        type: TYPES.INIT_NEW_RUNS,
+        payload: filteredInitRuns,
+      });
+
+      await dispatch({
+        type: TYPES.NEW_RUNS,
+        payload: result,
+      });
+      dispatch(constructToast("success", "Deleted!"));
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    dispatch(constructToast("danger", error?.response?.data?.message));
+    dispatch({ type: TYPES.NETWORK_ERROR });
+    dispatch({ type: TYPES.COMPLETED });
+  }
+};
```

Comment:
I think you can combine the two `dispatch()`'s for `TYPES.COMPLETED` if you move them outside the `try` block (e.g., at line 164).